src/shared/components/confirm_deletion.js

```diff
@@ -6,10 +6,7 @@ module.exports = React.createClass({
   },
 
   delete: function () {
-    const update_fn = this.props.payment
-      ? () => this.props.remove_payment(this.props.id)
-      : this.props.cancel_gift_aid
-    update_fn()
+    this.props.delete(this.props.id)
```

Comment:
It would probably be better to a) bind the passed in fns if they need an id or b) tweak the other functions to explicitly ignore id. I think this is OK enough to merge though